src/core/tests/frontend/paddle/test_models/gen_scripts/generate_box_coder.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from save_model import saveModel
+import sys
+
+
+def box_coder(name: str, prior_box, prior_box_var, target_box, code_type, box_normalized, axis, dtype):
+    import paddle
+    import paddle.fluid as fluid
+    paddle.enable_static()
+    is_tensor = not isinstance(prior_box_var, list)
+    with paddle.static.program_guard(paddle.static.Program(), paddle.static.Program()):
+
+        prior_box_decode = paddle.static.data(name='prior_box',
+                                              shape=prior_box.shape,
+                                              dtype=dtype)
+        target_box_decode = paddle.static.data(name='target_box',
+                                               shape=target_box.shape,
+                                               dtype=dtype)
+        if is_tensor:
+            prior_box_var_decode = paddle.static.data(name='prior_box_var',
+                                                      shape=prior_box_var.shape,
+                                                      dtype=dtype)
+        else:
+            prior_box_var_decode = prior_box_var
+
+        out = fluid.layers.box_coder(prior_box=prior_box_decode,
+                                     prior_box_var=prior_box_var_decode,
+                                     target_box=target_box_decode,
+                                     code_type=code_type,
+                                     box_normalized=box_normalized,
+                                     axis=axis)
+
+        cpu = paddle.static.cpu_places(1)
+        exe = paddle.static.Executor(cpu[0])
+        # startup program will call initializer to initialize the parameters.
+        exe.run(paddle.static.default_startup_program())
+
+        feed_dict = {'prior_box': prior_box, 'target_box': target_box}
+        if is_tensor:
+            feed_dict['prior_box_var'] = prior_box_var
+
+        outs = exe.run(
+            feed=feed_dict,
+            fetch_list=[out])
+
+        saveModel(name, exe, feedkeys=[*feed_dict.keys()], fetchlist=[out], inputs=[*feed_dict.values()], outputs=[outs[0]],
+                  target_dir=sys.argv[1])
+
+    return outs[0]
+
+
+def main():
+    # For decode
+    datatype = "float32"
+    prior_box = np.random.random([8, 4]).astype(datatype)
+    target_box = np.random.random([8, 4, 4]).astype(datatype)
+    prior_box_var = [0.1, 0.1, 0.1, 0.1]
+    code_type = "decode_center_size"
+    box_normalized = True
+    axis = 1
+    paddle_out = box_coder("box_coder_1", prior_box, prior_box_var, target_box, code_type, box_normalized, axis,
+                           datatype)
+
+    axis = 0
+    prior_box = np.random.random([4, 4]).astype(datatype)
+    box_normalized = False
+    prior_box_var = np.repeat(np.array([[0.1, 0.1, 0.1, 0.1]], dtype=np.float32), prior_box.shape[0], axis=0)
+    paddle_out = box_coder("box_coder_2", prior_box, prior_box_var, target_box, code_type, box_normalized, axis,
```

Comment:
Please also add a test case when "prior_box_var" is None.

Reply:
This "prior_box_var" could be of dimensions 1, 2, and 3. Can you please have all of them tested?

ref: https://github.com/PaddlePaddle/Paddle/blob/1f7e9546c5d2ae18c8d7588e62a30fc704a97f5a/python/paddle/fluid/tests/unittests/test_box_coder_op.py#L38

Reply:
Hi,
1. The "prior_box_var" can not be None, due to https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/detection.py#L952-L967

3. The "prior_box_var" can not be dimensions 3, in the first if branch, if "prior_box_var" is 2-d, it will be reshaped to 3-d
  ```python
    if pb_v.ndim == 2:
        var_shape = (1, pb_v.shape[0],
                     pb_v.shape[1]) if axis == 0 else (pb_v.shape[0], 1,
                                                       pb_v.shape[1])
    pb_v = pb_v.reshape(var_shape)
```
So the next if branch is to process 1-d and 2-d  "prior_box_var"
```python
    if pb_v.ndim == 1:
        tb_x = pb_v[0] * t_box[:, :, 0] * pb_w + pb_x
        tb_y = pb_v[1] * t_box[:, :, 1] * pb_h + pb_y
        tb_w = np.exp(pb_v[2] * t_box[:, :, 2]) * pb_w
        tb_h = np.exp(pb_v[3] * t_box[:, :, 3]) * pb_h
    else:
        tb_x = pb_v[:, :, 0] * t_box[:, :, 0] * pb_w + pb_x
        tb_y = pb_v[:, :, 1] * t_box[:, :, 1] * pb_h + pb_y
        tb_w = np.exp(pb_v[:, :, 2] * t_box[:, :, 2]) * pb_w
        tb_h = np.exp(pb_v[:, :, 3] * t_box[:, :, 3]) * pb_h
```

Reply:
Thanks for finding the discrepancy. Then it is fine to go with 1D and 2D prior_box_var. Make sure they are tested.